lib/Targets/AIETargets.cpp

```diff
@@ -153,6 +177,31 @@ void registerAIETranslations() {
       },
       registerDialects);
 
+  TranslateFromMLIRRegistration registrationShimDMAToJSON(
+      "aie-generate-json", "Transform AIE shim DMA allocation info into JSON",
+      [](ModuleOp module, raw_ostream &output) {
+        llvm::json::Object moduleJSON;
+        for (auto shimDMA_meta : module.getOps<ShimDMAAllocationOp>()) {
+          llvm::json::Object shimJSON;
+          ShimDMAAllocationOpAdaptor shimDMAAllocationOpAdaptor(shimDMA_meta);
+          auto channelDir = shimDMAAllocationOpAdaptor.getChannelDirAttr();
```

Comment:
I don't think you need the extra adaptor here. `shimDMA_meta.getChannelDirAttr();` should work.

Reply:
I accidentally merged in the PR, but I will resolve this in a new fix PR.